### PR TITLE
Added Discord Integrations Basics

### PR DESCRIPTION
**Pull request by nemodesm**

### CODE REVIEW

File: Assets/Scripts/Managers/TaskManager.cs
Comment (Taliayaya) [inline Assets/Scripts/Managers/TaskManager.cs:6]: You could use the Singleton class instead
```suggestion
	internal class TaskManager : Singleton<TaskManager>
```

File: Assets/Scripts/Managers/TaskManager.cs
Comment (Taliayaya) [inline Assets/Scripts/Managers/TaskManager.cs:93]: Hence making this part smaller, the singleton class Instance create the instance like you did if its not existing
```suggestion
		public static TaskState CreateTask(IEnumerator coroutine)
		{
		       if (!Instance)
		              Debug.Log("Creating instance...)
			return new TaskState(coroutine);
		}
```

File: Assets/Scripts/Managers/TaskManager.cs
Comment (Taliayaya) [inline Assets/Scripts/Managers/TaskManager.cs:40]:
```suggestion
				Instance.StartCoroutine(CallWrapper());
```

File: ProjectSettings/ProjectSettings.asset
Comment (Taliayaya) [inline ProjectSettings/ProjectSettings.asset:799]: I don't know what it does but is it ok if there is no longer UNITY_POST_PROCESSING_STACK_V2?

File: Assets/Scenes/SampleScene.unity
Comment (Taliayaya) [inline Assets/Scenes/SampleScene.unity:1]: There are lot of changes in this scene, is it important/in a prefab? Because when merging there may have conflicts

File: Assets/Scripts/Constants.cs
Comment (Taliayaya) [inline Assets/Scripts/Constants.cs:38]: Probably a better way to handler this indeed. we should migrate toward these in the future

**Review by Taliayaya — CHANGES_REQUESTED** [review]
Seems ok, maybe changing the TaskManager to what was said? 

Otherwise its good ig
you also name the Task class Task, I made one on my branch for the quests' task, but you have priority over this one, idk if we will keep both or rename mine.

File: Assets/Scenes/SampleScene.unity
Comment (nemodesm) [reply Assets/Scenes/SampleScene.unity:1]: The scene changes were only changing the titles to "= MASTERS" rather than "======MASTERS======"(as an example) so that the formatting worked, they can be ignored if needed (I thought i had not commited them anyways)

File: ProjectSettings/ProjectSettings.asset
Comment (nemodesm) [reply ProjectSettings/ProjectSettings.asset:799]: UNITY_POST_PROCESSING_STACK_V2 is for the biRP, this is using URP which has post processing built in, meaning (unless it has a safety check) using anything from UNITY_POST_PROCESSING_STACK_V2 will make the world purple because of error shaders

File: Assets/Scripts/Constants.cs
Comment (nemodesm) [reply Assets/Scripts/Constants.cs:38]: I just saw a dozen hard coded EventManager.AddListener("OnPause", ___) and figured a single typo somewhere could create a hard to find bug

Comment (Taliayaya) [reply Assets/Scripts/Constants.cs:38]: at first I wanted to use an enum but figured out it would create a lot of conflict as well. I guess const class is better indeed

File: Assets/Scripts/Managers/TaskManager.cs
Comment (nemodesm) [reply Assets/Scripts/Managers/TaskManager.cs:6]: I hadnt seen that the singleton could create itself, then the only thing that is stopping me from immediately switching to that is a little bug that singleton has whereby its not a singleton
image

File: Assets/Scripts/Constants.cs
Comment (nemodesm) [reply Assets/Scripts/Constants.cs:38]: the structure might need a little work though, i didnt want to go crazy on the class nesting (which i use as "folders")

File: Assets/Scripts/Managers/TaskManager.cs
Comment (Taliayaya) [reply Assets/Scripts/Managers/TaskManager.cs]: What kind of bug is it, is it multiplying itself? (if yes, I haven't seen it happen and still can't) Or the fact that multiple singleton are stacked on the same go?

Comment (nemodesm) [reply Assets/Scripts/Managers/TaskManager.cs]: When reloading the scene, it just creates a new one because it is a physical scene object and it doesn't check on awake if there is already one so if

created one (entered playmode)
access it (so that _instance is set)
create as many unregistered ones as you want

thats why i usually go for the [RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.BeforeSceneLoad)] method, but a check in awake also works, just with a little extra overhead

I can add the check in awake if you want cause i need to move it out of DefaultNamespace anyways cause it was causing issues with commits [786a3b7](https://github.com/Taliayaya/Project-86/pull/39/commits/786a3b7c847c8e84689ab8ca98aaba183379f8c8), [20671d7](https://github.com/Taliayaya/Project-86/pull/39/commits/20671d767be18ff7ac152cae1f1e217b66fa9f26), and
 [8aae355](https://github.com/Taliayaya/Project-86/pull/39/commits/8aae3554d3ff758a682ef3478d4d56f0ed8518ee)

Comment (Taliayaya) [reply Assets/Scripts/Managers/TaskManager.cs]: Oh that is probably explaining some bugs then. I never paid attention to be honest. You can modify it. If you want you can also remove the OnApplicationQuit that returns null instead of the instance. 
It makes more sense now.

File: Assets/Scenes/SampleScene.unity
Comment (nemodesm) [reply Assets/Scenes/SampleScene.unity:1]: Reverted

**Review by Taliayaya — APPROVED** [review]
Good to go
Any last word / action before merge?

**Comment by nemodesm** [comment]
I can't merge cause I don't have write access, you'll have to do it yourself @Taliayaya

**Comment by Taliayaya** [comment]
I'll take care of it

**Pull request merged**
